src/sql-parser/src/parser.rs

```diff
@@ -1564,19 +1554,35 @@ impl<'a> Parser<'a> {
                 vec![]
             };
             AvroSchema::Schema {
-                schema,
+                schemas,
                 with_options,
             }
         } else {
             return self.expected(
                 self.peek_pos(),
-                "CONFLUENT SCHEMA REGISTRY or SCHEMA",
+                "'CONFLUENT SCHEMA REGISTRY', 'KEY SCHEMA', or 'VALUE SCHEMA'",
                 self.peek_token(),
             );
         };
         Ok(avro_schema)
     }
 
+    fn parse_schema_pair(&mut self) -> Result<SchemaPair, ParserError> {
+        let key_schema = if self.parse_keyword(KEY) {
+            Some(self.parse_schema()?)
+        } else {
+            None
+        };
+        if !self.parse_keyword(VALUE) {
+            warn!("Avro schemas will require the syntax 'VALUE SCHEMA' in a future release");
```

Comment:
The chance that this warning is seen by the person who actually has the capacity to fix the query is somewhat slim. I guess this is better than nothing but it's too bad we can't easily send back a notice response to the client. 

I think we could at least make this message a bit more specific; maybe something like "the `FORMAT AVRO SCHEMA ...` syntax in `CREATE SOURCE` and `CREATE SINK` is deprecated; a future release will require that you explicitly indicate the schema types, as in `FORMAT AVRO [KEY SCHEMA ...] VALUE SCHEMA ...`.

Reply:
I agree a notice would be better, do we have any infrastructure for that?

Reply:
This is also an attempt to combat the problem of warn-level visibility: https://github.com/MaterializeInc/materialize/pull/6311